dev/Breadcrumb/Breadcrumb.cpp

```diff
@@ -0,0 +1,561 @@
+                m_notifyCollectionChanged.set(incc);
+            }
+
+            if (const auto& collection = itemsSource.try_as<winrt::IObservableVector<IInspectable>>())
+            {
+                m_itemsSourceChanged2 = collection.VectorChanged(winrt::auto_revoke, { this, &Breadcrumb::OnRepeaterCollectionChanged });
+            }
+        }
+    }
+}
+
+void Breadcrumb::OnRepeaterCollectionChanged(const winrt::IInspectable&, const winrt::IInspectable& args)
+{
+    if (const auto& breadcrumbItemRepeater = m_breadcrumbItemRepeater.get())
+    {
+        auto newItemsSource = GenerateInternalItemsSource();
+        breadcrumbItemRepeater.ItemsSource(newItemsSource);
+        breadcrumbItemRepeater.UpdateLayout();
+
+        // For some reason, when interacting with keyboard, the last element doesn't raise the OnPrepared event
+        ForceUpdateLastElement();
+    }
+
+    return;
+}
+
+void Breadcrumb::ForceUpdateLastElement()
+{
+    const auto& itemsSourceAsList = this->ItemsSource().as<winrt::Collections::IVector<winrt::IInspectable>>();
+    const uint32_t itemCount = itemsSourceAsList.Size();
+
+    if (const auto& breadcrumbRepeater = m_breadcrumbItemRepeater.get())
+    {
+        const auto& newLastItem = breadcrumbRepeater.TryGetElement(itemCount).try_as<winrt::BreadcrumbItem>();
+        UpdateLastElement(newLastItem);
+    }
+}
+
+void Breadcrumb::UpdateLastElement(const winrt::BreadcrumbItem& newLastBreadcrumbItem)
+{
+    // If the element is the last element in the array,
+    // then we reset the visual properties for the previous
+    // last element
+    if (const auto& lastItem = m_lastBreadcrumbItem.get())
+    {
+        auto lastItemImpl = winrt::get_self<BreadcrumbItem>(lastItem);
+        lastItemImpl->ResetVisualProperties();
+    }
+
+    if (const auto& newLastItemImpl = winrt::get_self<BreadcrumbItem>(newLastBreadcrumbItem))
+    {
+        newLastItemImpl->SetPropertiesForLastNode();
+        m_lastBreadcrumbItem.set(newLastBreadcrumbItem);
+    }
+}
+
+void Breadcrumb::OnElementPreparedEvent(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementPreparedEventArgs& args)
+{
+    if (const auto& item = args.Element().try_as<winrt::BreadcrumbItem>())
+    {
+        if (const auto& itemImpl = winrt::get_self<BreadcrumbItem>(item))
+        {
+            // The first element is always the ellipsis item
+            itemImpl->SetItemsRepeater(*this);
+
+            const uint32_t itemIndex = args.Index();
+            if (itemIndex == 0)
+            {
+                itemImpl->SetPropertiesForEllipsisNode();
+                itemImpl->SetFlyoutDataTemplate(DropdownItemTemplate());
+
+                m_ellipsisBreadcrumbItem.set(item);
+            }
+            else
+            {
+                const auto& itemsSourceAsList = this->ItemsSource().as<winrt::Collections::IVector<winrt::IInspectable>>();
```

Comment:
Same problem as in Breadcrumb::ForceUpdateLastElement():   Breadcrumb.ItemsSource is not necessarily an IVector. Did you mean to use the internal ItemsRepeater ItemsSource?